service/policy/db/resource_mapping.go

```diff
@@ -2,18 +2,171 @@ package db
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"log/slog"
 
 	sq "github.com/Masterminds/squirrel"
+	"github.com/google/uuid"
 	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgtype"
 	"github.com/opentdf/platform/protocol/go/common"
 	"github.com/opentdf/platform/protocol/go/policy"
 	"github.com/opentdf/platform/protocol/go/policy/resourcemapping"
 	"github.com/opentdf/platform/service/logger"
 	"github.com/opentdf/platform/service/pkg/db"
+	"github.com/opentdf/platform/service/pkg/util"
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+/*
+ Resource Mapping CRUD
+
+ NOTE: uses sqlc instead of squirrel
+*/
+
+func (c PolicyDBClient) ListResourceMappingGroups(ctx context.Context, r *resourcemapping.ListResourceMappingGroupsRequest) ([]*policy.ResourceMappingGroup, error) {
+	list, err := c.Queries.ListResourceMappingGroups(ctx, r.GetNamespaceId())
+	if err != nil {
+		return nil, err
+	}
+
+	resourceMappingGroups := make([]*policy.ResourceMappingGroup, len(list))
+
+	for i, rmGroup := range list {
+		resourceMappingGroups[i] = &policy.ResourceMappingGroup{
+			Id:          rmGroup.ID,
+			NamespaceId: rmGroup.NamespaceID,
+			Name:        rmGroup.Name,
+		}
+	}
+
+	return resourceMappingGroups, nil
+}
+
+func (c PolicyDBClient) ListResourceMappingGroupsByFqns(ctx context.Context, fqns []string) (map[string]*policy.ResourceMappingGroup, error) {
+	if (fqns == nil) || (len(fqns) == 0) {
+		return nil, fmt.Errorf("FQN list provided is empty")
```

Comment:
This check is now done with protovalidate and has since been [removed by that commit](https://github.com/opentdf/platform/pull/1343/commits/67ad6ea70ade6fd4065417a156db499c9f923b30)